t/op/taint.t

```diff
@@ -2525,7 +2524,7 @@ is eval { eval $::x.1 }, 1, 'reset does not taint undef';
     $ENV{PATH} = $old_env_path if $Is_MSWin32;
     is runperl(
        switches => [ '-T' ],
-       prog => 'use constant K=>$^X; 0 if K; BEGIN{} use strict; '
+       prog => 'use constant K=>$^X; 0 if K; BEGIN{} use Getopt::Long; '
```

Comment:
I'm not sure about this change? what's the reason?

Reply:
> I'm not sure about this change? what's the reason?

In https://github.com/atoomic/perl/issues/220#issuecomment-708531624, @brainbuz wrote:

> taint.t had a global stricture and in a runperl block was testing the 'use' command with -T, in the PR I propose changing 'use strict' to 'use Getopt::Long', any Core module would serve the intention of the test.

So he appears to feel that using `strict` as the module to test the `-T` switch is no longer the best choice.

I'm not sure I agree that we need to make a change here, but if we decide to do so, I'd rather use a module that, say, lives under `lib` or `ext` rather than one that is dual-life.

Thank you very much.
Jim Keenan

Reply:
I have reviewed the [original bug ticket in RT](https://github.com/Perl/perl5/issues/14059) that gave rise to line 2528 above, as well as the [commit](https://github.com/Perl/perl5/commit/64ff300be0f7714585466af5bb87b2e37db5082a) that introduced that line and the unit test in which that line is found.  The point of that line can be illustrated with this program:
```
$ cat rt-122669.pl 
#!perl -T
# https://github.com/Perl/perl5/commit/64ff300be0f7714585466af5bb87b2e37db5082a

# tainted constant
use constant K=>$^X;

# Just reading the constant for the sake of folding can enabled
# taintedness at compile time.
0 if K;

# Taintedness is still on when the ‘strict.pm’ SV is created, so
# require croaks on it (‘Insecure dependency’).
use strict;
#use less 'please';
```
... which is extracted from Father C's commit message in the commit cited.

Run this program, first in perl-5.18 and then in perl-5.20 (or any later version, including a `perl` built in our `alpha` branch).  The earlier version produced the surprising exception about which the original RT was concerned.
```
$ perl -v | head -2 | tail -1
This is perl 5, version 18, subversion 4 (v5.18.4) built for x86_64-linux

$ perl -T rt-122669.pl 
Insecure dependency in require while running with -T switch at rt-122669.pl line 13.
BEGIN failed--compilation aborted at rt-122669.pl line 13.
```
If, in the program above, you comment out `use strict;` and substitute any other `use` statement, you get exactly the same results in perl-5.18 versus later versions.  It's the `use` that is important here, not what is being used.

I think if we changed this to something *other than* strict, generations to come would be puzzled as to why we had done so.  So I recommend leaving the line as is -- which means that there's nothing left to be altered via this p.r. -- which means the p.r. is closable without merging.

I will do so in 3 days time unless I hear otherwise.

Thank you very much.
Jim Keenan